compiler/rustc_trait_selection/src/traits/error_reporting/type_err_ctxt_ext.rs

```diff
@@ -984,7 +984,10 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
             // Already reported in the query.
             SelectionError::NotConstEvaluatable(NotConstEvaluatable::Error(guar)) |
             // Already reported.
-            Overflow(OverflowError::Error(guar)) => return guar,
+            Overflow(OverflowError::Error(guar)) => {
+                self.set_tainted_by_errors(guar);
```

Comment:
Had to taint context here to stop the ICE.

Reply:
honestly, I'm fine regressing that one and reopening the issue if it gives you any trouble. It was fixed by accident last time.

Reply:
I have let it regress. Will re-open https://github.com/rust-lang/rust/issues/68830 when this PR is merged.